src/views/Users/ViewUser/index.jsx

```diff
@@ -115,79 +116,84 @@ function ViewUser({ isNewUser, ...props }) {
-          if (sc.change_type === 'insert') {
-            const p = getEmptyPermission();
-
-            p.name = sc.permission;
-            p.sc = clone(sc);
-            p.sc.name = sc.permission;
-            delete p.sc.permission;
-            permissions.push(p);
-          }
-        });
+            }));
+      const permissions =
+        userdata === null
+          ? []
+          : Object.keys(userdata.data.data.permissions).map(name => {
+              const details = userdata.data.data.permissions[name];
+              const sc = scheduledChanges.data.data.scheduled_changes.filter(
+                sc => sc.username === existingUsername && sc.permission === name
+              );
+              const permission = {
+                name,
+                options: details.options || { products: [], actions: [] },
+                data_version: details.data_version,
+                sc: null,
+                metadata: {
+                  isAdditional: false,
+                  productText: '',
+                  actionText: '',
+                },
+              };
+
+              // Copy any scheduled changes associated with an existing
+              // permission into the list of permissions.
+              if (sc.length > 0) {
+                // The backend refers to permission names as the 'permission'
+                // attribute of an object, but we prefer to call it 'name', so
+                // we have to adjust that.
+                sc[0].name = sc[0].permission;
+                delete sc[0].permission;
+
+                // Array destructuring is not very readable here.
+                // eslint-disable-next-line prefer-destructuring
+                permission.sc = sc[0];
+              }
+
+              return permission;
+            });
+
+      // Scheduled inserts don't have an existing permission to associate
+      // with, so we need to create an empty one, and add to it.
+      scheduledChanges.data.data.scheduled_changes.forEach(sc => {
+        if (sc.change_type === 'insert' && sc.username === existingUsername) {
+          const p = getEmptyPermission();
+
+          p.name = sc.permission;
+          p.sc = clone(sc);
+          p.sc.name = sc.permission;
+          delete p.sc.permission;
+          permissions.push(p);
+        }
+      });
 
+      if (userdata) {
         setUsername(userdata.data.data.username);
-        setRoles(roles);
-        setOriginalRoles(clone(roles));
-        setPermissions(permissions);
-        setOriginalPermissions(clone(permissions));
-        setProducts(productdata.data.data.product);
-        setRequiredSignoffs(rs.data.data.required_signoffs);
-      });
-    }
+      }
+
+      setRoles(roles);
+      setOriginalRoles(clone(roles));
+      setPermissions(permissions);
+      setOriginalPermissions(clone(permissions));
+      setProducts(productdata.data.data.product);
+      setRequiredSignoffs(rs.data.data.required_signoffs);
+    });
   }, []);
```

Comment:
Navigating to http://localhost:9000/users/balrogadmin then selecting a product restriction for the last permission with the name "rule" throws an error in the console `TypeError: result.sc.options is null` and then shows a blank page. Works for the product restriction of "release_read_only" however.

Reply:
Is that permission currently a scheduled change, or a full fledged permission? There's probably something busted in the `useEffect`, but I'm having trouble reproducing.

Reply:
![Screen Shot 2019-07-31 at 4 15 53 PM](https://user-images.githubusercontent.com/3766511/62244966-80513d80-b3ae-11e9-9310-e23e26beb7bf.png)

Reply:
Hm, I think after my recent changes it's no longer possible to get into this state. However, I'm going to push a fix that will ensure that even if sc.options from the server is null, it doesn't end up that way in the state.